### PR TITLE
clrfund/monorepo <- ethereum/clrfund

### LINKED ISSUE #58: Create subgraph

### LINKED ISSUE #343: Create UI for optimistic recipient registry to remove projects
Similar to https://github.com/clrfund/monorepo/issues/295, it'd be nice to have a UI for users to request removal of recipients.

- Ability to submit `removeRecipient` tx to the registry contract, e.g. a "Report this project" button on the project page.
- Ability to share evidence with the registry owner (to either prove you are the recipient & want to be removed or that the recipient is somehow breaking the rules, e.g. by bribing contributors). This evidence is needed for the owner to determine if the removal request is legitimate. This could be a form or simple instructions to email the evidence somewhere / report the evidence on the CLR.fund forum.

### LINKED ISSUE #349: Add support for more wallets
Currently (from my understanding) only EIP-1193 providers are supported (https://github.com/clrfund/monorepo/pull/99).

It'd be useful to add a tool like Wallet Connect (e.g. using [`web3modal`](https://www.npmjs.com/package/web3modal)) to support mobile & other wallets.

### PR DESCRIPTION
**Pull request by wackerow**
Merging the `ethereum/clrfund` fork back into `clrfund/monorepo`. 

**Edit - @samajammin adding summary of changes:**

## Summary of changes 

- Product landing page: https://eth2clrfund.netlify.app/#/
- Educational content (10 about pages), customized based off network: https://eth2clrfund.netlify.app/#/about
- User onboarding flows
    - Dedicated landing pages for recipient (https://eth2clrfund.netlify.app/#/join) and contributor (https://eth2clrfund.netlify.app/#/verify) onboarding
    - Recipient application flow for the optimistic registry with expanded project data
        - Optional email input with form integration to Google Sheets (based on ENV variable)
    - BrightID flow for contributors with BrightID API integration
- Enhanced UX for full round lifecycle & user states
    - Store getters for computed properties across the app
    - Added relevant CTAs across the app depending on round & user state
    - Added a "join phase" for if no funding round exists
    - Countdown banners & contextual information depending on current phase
- Enhanced wallet integrations
    - Web3 plugin, with WalletConnect & MetaMask support
    - Prompts to switch networks
    - Auto prompt sign on refresh
- Cart features
    - Collapse / expand
    - Split evenly
    - Reallocation messages
- User profile view
    - Display ETH & token balance
- Enhanced project page view, based off new fields in recipient application
- Transaction success screens (for registry, contribution & reallocation txs)
- Subgraph integration
    - Replaced queryFilter for event logs
- Additional network & token support
    - Arbitrum support
    - Flexible content & UI depending on network & token used
- Mobile support across the app

### CODE REVIEW

**Comment by samajammin** [comment]
@daodesigner note: we recognize this is a monster PR 😅. I'm planning to add some details on the primary changes here. Will do this ASAP - planning to do it this week - but in Lisbon for a conference & things are a bit hectic.

**Comment by samajammin** [comment]
Ok @daodesigner - wasn't quite sure what level of detail to go into here but figured no sense in giving you the line-by-line or even the file-by-file changes. I've added this summary below but please let me know if you have more questions or would like to see more detail. I've also included preview deploy links where relevant.

# Summary of changes 

- Product landing page: https://eth2clrfund.netlify.app/#/
- Educational content (10 about pages), customized based off network: https://eth2clrfund.netlify.app/#/about
- User onboarding flows
    - Dedicated landing pages for recipient (https://eth2clrfund.netlify.app/#/join) and contributor (https://eth2clrfund.netlify.app/#/verify) onboarding
    - Recipient application flow for the optimistic registry with expanded project data
        - Optional email input with form integration to Google Sheets (based on ENV variable)
    - BrightID flow for contributors with BrightID API integration
- Enhanced UX for full round lifecycle & user states
    - Store getters for computed properties across the app
    - Added relevant CTAs across the app depending on round & user state
    - Added a "join phase" for if no funding round exists
    - Countdown banners & contextual information depending on current phase
- Enhanced wallet integrations
    - Web3 plugin, with WalletConnect & MetaMask support
    - Prompts to switch networks
    - Auto prompt sign on refresh
- Cart features
    - Collapse / expand
    - Split evenly
    - Reallocation messages
- User profile view
    - Display ETH & token balance
- Enhanced project page view, based off new fields in recipient application
- Transaction success screens (for registry, contribution & reallocation txs)
- Subgraph integration
    - Replaced queryFilter for event logs
- Additional network & token support
    - Arbitrum support
    - Flexible content & UI depending on network & token used
- Mobile support across the app

**Comment by samajammin** [comment]
@auryn-macmillan @daodesigner regarding the support for viewing multiple rounds (see discussion in https://github.com/ethereum/clrfund/issues/431) - @pettinarip is currently working on an implementation to add the funding round factory & funding round to the route params which should be a more robust solution than the current approach of deploying multiple instances of the application & referencing the different IPFS hashes. We'll keep you focused on that but in short, we believe we should keep that work separate from this merge PR given that it alone may add some complex changes. The application as-is keeps parity with the existing monorepo codebase.

File: vue-app/.env.example [hunk 1]
```diff
@@ -17,3 +25,11 @@ VUE_APP_RECIPIENT_REGISTRY_POLICY=QmeygKjvrpidJeFHv6ywjUrj718nwtFQgCCPPR4r5nL87R
 
 # Comma-separated list of IPFS hashes
 VUE_APP_EXTRA_ROUNDS=
+
+# Operator of clr.fund instance
+VUE_APP_OPERATOR=
+
+# Google Service Account credentials in JSON format
+GOOGLE_APPLICATION_CREDENTIALS=
```
Comment (daodesigner) [inline vue-app/.env.example:34]: Add to documentation how to get the Spreadsheet parameters

File: vue-app/.env.example [hunk 2]
```diff
@@ -17,3 +25,11 @@ VUE_APP_RECIPIENT_REGISTRY_POLICY=QmeygKjvrpidJeFHv6ywjUrj718nwtFQgCCPPR4r5nL87R
...
+# Spreadsheet ID to send recipients data
+VUE_APP_GOOGLE_SPREADSHEET_ID=
```
Comment (daodesigner) [inline vue-app/.env.example:36]: What does this track?

File: vue-app/.env.example [hunk 3]
```diff
@@ -9,6 +15,8 @@ VUE_APP_CLRFUND_FACTORY_ADDRESS=0x5FC8d32690cc91D4c39d9d3abcBD16989F875707
 
 # Supported values: simple, brightid
 VUE_APP_USER_REGISTRY_TYPE=simple
+# clr.fund (prod) or CLRFundTest (testing)
+VUE_APP_BRIGHTID_CONTEXT=clr.fund
```
Comment (daodesigner) [inline vue-app/.env.example:20]: Is there a context on Arbitrum that we can use set up already?

**Review by daodesigner — CHANGES_REQUESTED** [review]
Add Information to Docs regarding Google spread sheet tracking (very cool feature) and BrightID Context for Arbitrum if possible. Otherwise looks good, great work everyone!

File: vue-app/.env.example (hunk 3 above)
Comment (samajammin) [reply vue-app/.env.example:20]: The BrightID context is distinct from the Ethereum chain we use - it's specific to BrightID. I'll add some documentation around this.

**Comment by samajammin** [comment]
@daodesigner I've expanded the docs to address your change requests. You can see the specific changes here: https://github.com/ethereum/clrfund/pull/453

The Google Sheet integration is definitely a bit of an edge use case given the use of Netlify functions, so no hard feelings if you'd rather just rip out that feature entirely. It's disabled by default though so shouldn't cause any harm to leave in (other than more code/files).

Please let me know if you have any other questions or requests!

**Review by daodesigner — APPROVED** [review]

**Comment by auryn-macmillan** [comment]
its happening

**Comment by corwintines** [comment]
Oh baby!

**Pull request merged**